File: backend/src/main/java/com/java/NBE4_5_1_7/domain/study/dto/StudyContentDetailDto.java
package com.java.NBE4_5_1_7.domain.study.dto;

import com.java.NBE4_5_1_7.domain.study.entity.StudyContent;
import lombok.Getter;
import lombok.RequiredArgsConstructor;
import lombok.Setter;

@Getter
@Setter
@RequiredArgsConstructor
public class StudyContentDetailDto {
    private Long id;
    private String title;
    private String body;
    private String firstCategory;
    private String secondCategory;

    public StudyContentDetailDto(StudyContent studyContent) {
        this.id = studyContent.getStudy_content_id();
        this.title = studyContent.getTitle();
        this.body = studyContent.getBody();
        this.firstCategory = studyContent.getFirstCategory().name();
        this.secondCategory = studyContent.getSecondCategory();
    }
}

File: backend/src/main/java/com/java/NBE4_5_3_7/domain/study/dto/response/StudyContentDetailDto.kt
package com.java.NBE4_5_3_7.domain.study.dto.response

import com.java.NBE4_5_3_7.domain.study.entity.StudyContent

class StudyContentDetailDto {
    var id: Long? = null
    var title: String? = null
    var body: String? = null
    var firstCategory: String? = null
    var secondCategory: String? = null

    constructor(studyContent: StudyContent) {
        this.id = studyContent.study_content_id
        this.title = studyContent.title
        this.body = studyContent.body
        this.firstCategory = studyContent.firstCategory.name
        this.secondCategory = studyContent.secondCategory
    }
}
